AWSClientRuntime/Sources/Middlewares/EndpointResolverMiddleware.swift

```diff
@@ -19,51 +19,52 @@ public struct EndpointResolverMiddleware<OperationStackOutput: HttpResponseBindi
                           input: SdkHttpRequestBuilder,
                           next: H) async throws -> OperationOutput<OperationStackOutput>
     where H: Handler,
-    Self.MInput == H.Input,
-    Self.MOutput == H.Output,
-    Self.Context == H.Context {
-        
-        guard let region = context.getRegion(), !region.isEmpty else {
-            throw SdkError<OperationStackError>.client(ClientError.unknownError(("Region is unable to be resolved")))
-        }
-        
-        do {
-            let awsEndpoint = try endpointResolver.resolve(serviceId: serviceId,
-                                                           region: region)
-            var host = ""
-            if let overrideHost = context.getHost() {
-                host = overrideHost
-            } else {
-                host = "\(context.getHostPrefix() ?? "")\(awsEndpoint.endpoint.host)"
-            }
-            
-            if let protocolType = awsEndpoint.endpoint.protocolType {
-                input.withProtocol(protocolType)
-            }
-            
-            var updatedContext = context
-            
-            if let signingRegion = awsEndpoint.signingRegion {
-                updatedContext.attributes.set(key: HttpContext.signingRegion, value: signingRegion)
-            }
-            
-            if let signingName = awsEndpoint.signingName {
-                updatedContext.attributes.set(key: HttpContext.signingName, value: signingName)
-            }
-            
-            input.withMethod(context.getMethod())
-                .withHost(host)
-                .withPort(awsEndpoint.endpoint.port)
-                .withPath(context.getPath())
-                // TODO: investigate if this header should be the same host value as
-                // the actual host and where this header should be set
-                .withHeader(name: "Host", value: host)
-            
-            return try await next.handle(context: updatedContext, input: input)
-        } catch {
-            throw SdkError<OperationStackError>.client(ClientError.unknownError(("Endpoint is unable to be resolved")))
-        }
-    }
+          Self.MInput == H.Input,
+          Self.MOutput == H.Output,
+          Self.Context == H.Context {
+              
+              guard let region = context.getRegion(), !region.isEmpty else {
+                  throw SdkError<OperationStackError>.client(ClientError.unknownError(("Region is unable to be resolved")))
+              }
+              var awsEndpoint: AWSEndpoint
+              do {
+                  awsEndpoint = try endpointResolver.resolve(serviceId: serviceId,
+                                                             region: region)
```

Comment:
intentional change to endpoint handling? This code looks like it maintains the previous behavior?

Reply:
ya we weren't actually bubbling up the correct error before with the way the code was written. this is a minor change that keeps same beahvior but now the correct error gets bubbled up to the caller. with the previous code, every error looked like an endpoint error when in reality it wasn't :)